bundles/remote_services/discovery_zeroconf/src/discovery_zeroconf_announcer.c

```diff
@@ -115,38 +114,23 @@ celix_status_t discoveryZeroconfAnnouncer_create(celix_bundle_context_t *ctx, ce
     celix_autoptr(celix_thread_mutex_t) mutex = &announcer->mutex;
 
     celix_autoptr(celix_string_hash_map_t) endpoints = announcer->endpoints = celix_stringHashMap_create();
-    assert(announcer->endpoints != NULL);
+    if (endpoints == NULL) {
+        celix_logHelper_logTssErrors(logHelper, CELIX_LOG_LEVEL_ERROR);
+        celix_logHelper_fatal(logHelper, "Announcer: Failed to create endpoints map.");
```

Comment:
I think the log level should be error. This is what we currently describe in the development guide:

https://github.com/apache/celix/tree/dc0e0aebb282ce05b51631dd5c53b14783b856ba/documents/development#error-handling-and-logging

> error: This level should be used to report issues that need immediate attention and might prevent the system from functioning correctly. These are problems that are unexpected and affect functionality, but not so severe that the process needs to stop. Examples include runtime errors, inability to connect to a service, etc.
> fatal: Use this level to report severe errors that prevent the program from continuing to run. After logging a fatal error, the program will typically terminate.

Optionally we can also update the development guideline if we think situations like ENOMEM should be reported as fatal.

Reply:
IMO, `ENOMEM` should not be fatal.